DatadogInternal/Sources/NetworkInstrumentation/URLSession/URLSessionSwizzler.swift

```diff
@@ -0,0 +1,273 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+import Foundation
+
+internal class URLSessionSwizzler {
+    /// Counts of bindings of the URL swizzler.
+    ///
+    /// This value will increment for each call to the `bind()` method.
+    /// Calling `unbind()` will decrement the count, when reaching zero, the swizzler is disabled.
+    internal private(set) static var bindingsCount: Int = 0
```

Comment:
suggestion/ Including it in test integrity checks will ensure we don't go into flakiness because of missing or not balancing `unbind()` well in some tests:  https://github.com/DataDog/dd-sdk-ios/blob/bdd0cef7e50b2fc478583365781f357cad963c17/Tests/DatadogTests/TestsObserver/DatadogTestsObserver.swift#L23

Reply:
It is already checked with:
```swift
.init(
    assert: { activeSwizzlingNames.isEmpty },
    problem: "No swizzling must be applied.",
    solution: """
    Make sure all applied swizzling are reset by the end of test with `unswizzle()`.

    `DatadogTestsObserver` found \(activeSwizzlingNames.count) leaked swizzlings:
    \(activeSwizzlingNames.joined(separator: ", "))
    """
),
```